android/src/main/java/com/situm/plugin/SitumPluginImpl.java

```diff
@@ -255,33 +257,76 @@ public void invoke(Object... args) {
                 }
             };
 
-            final Callback onPermissionDenied = new Callback() {
+            final Callback onLocationPermissionDenied = new Callback() {
                 @Override
                 public void invoke(Object... args) {
                     Log.e(TAG, "Failed to request location permission.");
                 }
             };
 
-            Callback onPermissionCheckFailed = new Callback() {
+            Callback onLocationPermissionCheckFailed = new Callback() {
                 @Override
                 public void invoke(Object... args) {
 
                     Log.e(TAG, "Failed to check location permission.");
                 }
             };
 
-            Callback onPermissionChecked = new Callback() {
+            Callback onLocationPermissionChecked = new Callback() {
                 @Override
                 public void invoke(Object... args) {
                     boolean hasPermission = (boolean) args[0];
 
                     if (!hasPermission) {
-                        perms.requestPermission(Manifest.permission.ACCESS_FINE_LOCATION, new PromiseImpl(onPermissionGranted, onPermissionDenied));
+                        perms.requestPermission(Manifest.permission.ACCESS_FINE_LOCATION, new PromiseImpl(onLocationPermissionGranted, onLocationPermissionDenied));
                     }
                 }
             };
 
-            perms.checkPermission(Manifest.permission.ACCESS_FINE_LOCATION, new PromiseImpl(onPermissionChecked, onPermissionCheckFailed));
+            perms.checkPermission(Manifest.permission.ACCESS_FINE_LOCATION, new PromiseImpl(onLocationPermissionChecked, onLocationPermissionCheckFailed));
+
+            // Bluetooth permissions
+
+            final Callback onBluetoothPermissionsGranted = new Callback() {
+                @Override
+                public void invoke(Object... args) {
+                    String result = (String) args[0];
+                    if (!result.equals("granted")) {
+                        Log.e(TAG, "Nearby devices permission was not granted.");
```

Comment:
Non entendo o que fan estas callbacks, simplemente pintan un log de erro? Non se escala a callback? Algo me estou perdendo.

Reply:
Eu únicamente quería añadir o permiso de bluetooth para que funcionase o plugin do SDK en Android >= 12 e poder validalo.

Ademáis fuiseme a olla e pensaba que esta clase era unha dos exemplos, por eso metín tanto boilerplate code. Penso que como mínimo debería pedir todo de unha vez con **requestMultiplePermissions()** porque así hay que pulsar 2 veces (1 para o permiso de ubicación e outra para o de bluetooth).

O tema da callback entendo que interesa, pero recordemos que ahora en WYF estamos esperando a que o SDK nos avise de que faltan permisos (co LocationListener#onError()) para pedílos. Entonces penso que sería máis ben desta segunda forma certo ? (ou incluso deixalo en mans do integrador, paréceme algo máis a nivel app para que o integrador faga e desfaga)